Modules/Sources/JetpackStatsWidgetsCore/HomeWidgetData.swift

```diff
@@ -0,0 +1,13 @@
+import Foundation
+
+public protocol HomeWidgetData: Codable {
+
+    var siteID: Int { get }
+    var siteName: String { get }
+    var url: String { get }
+    var timeZone: TimeZone { get }
+    var date: Date { get }
+    var statsURL: URL? { get }
+
+    static var filename: String { get }
+}
```

Comment:
It's not clear to me why this needs to be a `protocol` instead of a `struct`. It's possible the reason has to do with the generics usage, or it might be that the generics are due to this being a protocol and not a plain Swift object. It seemed out of scope for this PR to investigate further.